build/js/main.js

```diff
@@ -1177,89 +1199,80 @@ class Bluetooth extends Plugin {
-        if (val.Name === '')
-            this.status(`Pairing with ${val.Address}`);
+        var idx = this.discoveredDeviceList.selectedIndex;
+        if (this.discoveredDevices[idx].name === "")
+            this.status(`Pairing to ${this.discoveredDevices[idx].address}`);
         else
-            this.status(`Pairing with ${val.Name}`);
+            this.status(`Pairing to ${this.discoveredDevices[idx].name}`);
 
-        api.putPlugin(this.callsign, 'Pair', '{"Address" : "' + val.Address + '"}', (err, resp) => {
-            if (err !== null) {
+        api.putPlugin(this.callsign, 'Pair', '{"address" : "' + this.discoveredDevices[idx].address + '"}', (err,resp) =>{
+
+        if (err !== null) {
                 console.error(err);
                 return;
             }
-
-            // update Paired device list after 2s
-            setTimeout(this.getPairedDevices.bind(this), 2000);
-        });
+            document.getElementById('BT_Pair').disabled = true;
+            setTimeout(this.update.bind(this), 1000);
+            document.getElementById('BT_Connect').disabled = false;
+            });
     }
 
     connect() {
-        var idx = this.devicesListEl.selectedIndex;
+        var idx = this.pairedDeviceList.selectedIndex;
+        if (this.pairedDevices[idx].name === "")
+            this.status(`Connecting to ${this.pairedDevices[idx].address}`);
+        else
+            this.status(`Connecting to ${this.pairedDevices[idx].name}`);
+
 
-        this.status(`Connecting to ${this.pairedDevices[idx].Name}`);
-        api.putPlugin(this.callsign, 'Connect', '{"Address" : "' + this.pairedDevices[idx].Address + '"}', (err,resp) =>{
+        api.putPlugin(this.callsign, 'Connect', '{"address" : "' + this.pairedDevices[idx].address + '"}', (err,resp) =>{
             if (err !== null) {
                 console.error(err);
                 return;
             }
-
-            // update after 2s
-            setTimeout(this.update.bind(this), 2000);
+            setTimeout(this.update.bind(this), 1000);
+            document.getElementById('BT_Disconnect').disabled = false;
         });
     }
 
     disconnect() {
-        this.status(`Disconnecting to ${this.connected}`);
-        api.deletePlugin(this.callsign, 'Connect', null, (err,resp) =>{
-            if (err !== null) {
-                console.error(err);
-                return;
-            }
+        var idx = this.connectedDeviceList.selectedIndex;
+        if (this.connectedDevices[idx].name === "")
+            this.status(`Disconnecting to ${this.connectedDevices[idx].address}`);
+        else
+            this.status(`Disconnecting to ${this.connectedDevices[idx].name}`);
 
-            // update after 2s
-            setTimeout(this.update.bind(this), 2000);
+        var body = '{"address"  : "' + this.connectedDevices[idx].address + '"}';
+        api.deletePlugin(this.callsign, 'Connect', body, (err,resp) =>{
```

Comment:
same here, if there is JSONRPC available consider using `api.req` for backwards/forward compatibility.